src/app/application.cpp

```diff
@@ -516,9 +516,9 @@ int Application::exec(const QStringList &params)
 #endif // DISABLE_GUI
 #endif // DISABLE_WEBUI
 
-    new SearchPluginManager;
-    new RSS::Session; // create RSS::Session singleton
     new RSS::AutoDownloader; // create RSS::AutoDownloader singleton
+    new RSS::Session; // create RSS::Session singleton
```

Comment:
Please don't produce unnecessary changes (especially where you don't know)! RSS::AutoDownloader depends from RSS::Session so it should be created after RSS::Session.

Reply:
And more... In most cases, the application components must be removed in reverse order. So if you create it:
```c++
new RSS::Session; // create RSS::Session singleton
new RSS::AutoDownloader; // create RSS::AutoDownloader singleton
new SearchPluginManager;
```
you should remove it:
```c++
delete SearchPluginManager::instance();
delete RSS::AutoDownloader::instance();
delete RSS::Session::instance();
```

Reply:
I didn't understand the implications of changing this. I'll change it back.

Reply:
>RSS::AutoDownloader depends from RSS::Session so it should be created after RSS::Session.

This means that RSS::AutoDownloader uses RSS::Session and assumes that it already exists. So we need create RSS::Session before RSS::AutoDownloader and delete in reverse order (RSS::AutoDownloader can call RSS::Session methods in its constructor and destructor).